sdk/eventhub/Azure.Messaging.EventHubs/tests/Compatibility/TrackOneExceptionExtensionsTests.cs

```diff
@@ -0,0 +1,102 @@
+        ///   method.
+        /// </summary>
+        ///
+        [Test]
+        public void MapExceptionValidatesTheInstance()
+        {
+            Assert.That(() => ((TrackOne.EventHubsException)null).MapException(), Throws.ArgumentNullException);
+        }
+
+        /// <summary>
+        ///   Verifies that well-known derrived exception types have the correct value for their
+        ///   <see cref="EventHubsException.IsTransient" /> property.
+        /// </summary>
+        ///
+        [Test]
+        [TestCaseSource(nameof(ExceptionMappingTestCases))]
+        public void DerrivedExceptionsHaveTheCorrectTransientValues(Exception exception,
+                                                                    Type expectedMappedType)
+        {
+            var eventHubsException = (TrackOne.EventHubsException)exception;
+            var mappedException = eventHubsException.MapException();
+
+            Assert.That(mappedException, Is.Not.Null, "The mapping should produce an exception.");
+            Assert.That(mappedException.GetType(), Is.EqualTo(expectedMappedType), "The mapped exception type was incorrect.");
+            Assert.That(mappedException.IsTransient, Is.EqualTo(eventHubsException.IsTransient), "The mapped exception should agree on being transient.");
+            Assert.That(mappedException.ResourceName, Is.EqualTo(eventHubsException.EventHubsNamespace), "The mapped exception should use the namespace as its resource name.");
+            Assert.That(mappedException.Message, Does.Contain(eventHubsException.EventHubsNamespace ?? String.Empty), "The mapped exception should include the namespace in its message.");
+            Assert.That(mappedException.Message, Does.Contain(eventHubsException.RawMessage), "The mapped exception should include the message text in its message.");
+            Assert.That(mappedException.InnerException, Is.EqualTo(eventHubsException), "The mapped exception should wrap the original instance.");
+        }
+
+        /// <summary>
+        ///   Verifies that derrived exception types in the current library are well-known and have a
+        ///   corresponding test case.
+        /// </summary>
+        ///
+        [Test]
+        public void EventHubsExceptionTypesShouldHaveMappings()
+        {
+            var allDerrivedTypes = typeof(EventHubsException)
+               .Assembly
+               .GetTypes()
+               .Where(type => typeof(EventHubsException).IsAssignableFrom(type))
+               .Select(type => type.Name)
+               .OrderBy(name => name);
+
+            var knownDerrivedTypes = ExceptionMappingTestCases()
+                .Select(testCase => ((Type)testCase[1]).Name)
+                .OrderBy(name => name);
+
+            Assert.That(allDerrivedTypes, Is.EquivalentTo(knownDerrivedTypes), "All exceptions derrived from EventHubsException in the client library should have a matching mapping.");
+        }
+    }
+}
```

Comment:
Not appearing here, but the IDE shows a new line

Reply:
When there isn't a new-line, GitHub shows that ❌ icon at the last line.